projects/storefrontlib/src/cms-components/cart/add-to-cart/add-to-cart.component.ts

```diff
@@ -43,16 +55,41 @@ export class AddToCartComponent implements OnInit, OnDestroy {
     quantity: new FormControl(1, { updateOn: 'blur' }),
   });
 
+  constructor(
+    modalService: ModalService,
+    currentProductService: CurrentProductService,
+    cd: ChangeDetectorRef,
+    activeCartService: ActiveCartService,
+    componentData: CmsComponentData<model>
+  );
+
+  /**
+   * @deprecated since 3.4
+   */
+  constructor(
+    modalService: ModalService,
+    currentProductService: CurrentProductService,
+    cd: ChangeDetectorRef,
+    activeCartService: ActiveCartService
+  );
+
   constructor(
     protected modalService: ModalService,
     protected currentProductService: CurrentProductService,
-    protected cd: ChangeDetectorRef,
-    protected activeCartService: ActiveCartService
+    private cd: ChangeDetectorRef,
+    protected activeCartService: ActiveCartService,
+    @Optional() protected componentData?: CmsComponentData<model>
   ) {}
 
   ngOnInit() {
+    this.componentData?.data$.subscribe((data: model) => {
+      if (data.inventoryDisplay && data.inventoryDisplay == 'true') {
+        this.showInventory = true;
+      }
+    });
+
     if (this.product) {
-      this.productCode = this.product.code;
+      this.productCode = this.product.code ? this.product.code : '';
```

Comment:
```suggestion
      this.productCode = this.product.code ?? '';
```

Reply:
since you're fixing this as you added the union type with undefined, you should fix line 105 as well, which contains the same error

line 105 = this.productCode = product.code